controllers/confidential_handler.go

```diff
@@ -0,0 +1,71 @@
+package controllers
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func patchConfigMap(client client.Client, name string, namespace string, data map[string]string) error {
+	configMap := &corev1.ConfigMap{}
+	if err := client.Get(context.TODO(), types.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}, configMap); err != nil {
+		return err
+	}
+
+	for key, value := range data {
+		configMap.Data[key] = value
+	}
+
+	return client.Update(context.TODO(), configMap)
+}
+
+func (r *KataConfigOpenShiftReconciler) toggleFeatureConfidential(enabled bool) error {
+
+	var ImageConfigMapData map[string]string
+	var peerpodsCMData map[string]string
+
+	if enabled {
+		ImageConfigMapData = map[string]string{"CONFIDENTIAL_COMPUTE_ENABLED": "yes"}
+		peerpodsCMData = map[string]string{"DISABLECVM": "false"}
+	} else {
+		ImageConfigMapData = map[string]string{"CONFIDENTIAL_COMPUTE_ENABLED": "no"}
+		peerpodsCMData = map[string]string{"DISABLECVM": "true"}
+	}
+
+	// Changes to make before image creation job starts:
+	//   - Edit ImageConfigMap to create a confidential image.
+
+	ig := GetImageGenerator()
+
+	if enabled {
+		// Create ImageConfigMap, if it doesn't exist already.
+		if err := ig.createImageConfigMapFromFile(); err != nil {
```

Comment:
Should we think about cases where Confidential is supported but image generation isn't (libvirt?), i'm not sure it will go through successfully here
maybe we should consider confidential featureGating to supported providers only? (i.e. have a function to check provider is supported)